modules/twinklespeedy.js

```diff
@@ -1295,35 +1297,11 @@ Twinkle.speedy.callbacks = {
 				return;
 			}
 
-			var code, parameters, i;
-			if (params.normalizeds.length > 1) {
-				code = "{{db-multiple";
-				params.utparams = {};
-				$.each(params.normalizeds, function(index, norm) {
-					code += "|" + norm.toUpperCase();
-					parameters = params.templateParams[index] || [];
-					for (var i in parameters) {
-						if (typeof parameters[i] === 'string' && !parseInt(i, 10)) {  // skip numeric parameters - {{db-multiple}} doesn't understand them
-							code += "|" + i + "=" + parameters[i];
-						}
-					}
-					$.extend(params.utparams, Twinkle.speedy.getUserTalkParameters(norm, parameters));
-				});
-				code += "}}";
-			} else {
-				parameters = params.templateParams[0] || [];
-				code = "{{db-" + params.values[0];
-				for (i in parameters) {
-					if (typeof parameters[i] === 'string') {
-						code += "|" + i + "=" + parameters[i];
-					}
-				}
-				if (params.usertalk) {
-					code += "|help=off";
-				}
-				code += "}}";
-				params.utparams = Twinkle.speedy.getUserTalkParameters(params.normalizeds[0], parameters);
-			}
+			// given the params, builds the template and also adds the user talk page parameters to the params that were passed in
+			// returns => [<string> wikitext, <object> newParams]
+			var buildData = Twinkle.speedy.callbacks.getTemplateCodeAndParams(params),
+				code = buildData[0];
+			params = buildData[1];
```

Comment:
Since it's only modifying params.utparams  within the params object, just make it return [wikitext, utparams]. Better encapsulation that way, I think.